src/GPSPoint.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Zend\Validator;
+
+final class GPSPoint extends AbstractValidator
+{
+
+    const OUT_OF_BOUNDS = 'gpsPointOutOfBounds';
+    const CONVERT_ERROR = 'gpsPointConvertError';
+    const INCOMPLETE_COORDINATE = 'gpsPointIncompleteCoordinate';
+
+    /**
+     * @var array
+     */
+    protected $messageTemplates = [
+        'gpsPointOutOfBounds' => '%value% is out of Bounds.',
+        'gpsPointConvertError' => '%value% can not converted into a Decimal Degree Value.',
+        'gpsPointIncompleteCoordinate' => '%value% did not provided a complete Coordinate',
+    ];
+
+    /**
+     * Returns true if and only if $value meets the validation requirements
+     *
+     * If $value fails validation, then this method returns false, and
+     * getMessages() will return an array of messages that explain why the
+     * validation failed.
+     *
+     * @param  mixed $value
+     * @return bool
+     * @throws Exception\RuntimeException If validation of $value is impossible
+     */
+    public function isValid($value)
+    {
+        if (strpos($value, ',') === false) {
+            $this->error(GPSPoint::INCOMPLETE_COORDINATE, $value);
+            return false;
+        }
+
+        list($lat, $long) = explode(',', $value);
+
+        if ($this->isValidCoordinate($lat, 90.0000) && $this->isValidCoordinate($long, 180.000)) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @param string $value
+     * @param $maxBoundary
+     * @return bool
+     */
+    private function isValidCoordinate($value, $maxBoundary)
```

Comment:
Again, this is the direction we're *moving*. While we've defaulted to
protected in the *past*, we're changing practices based on the maintenance
issues the practice has created. (For your reference, I'm project lead; you
can take this as canonical.)
On Jul 13, 2015 5:12 PM, "dima" <notifications@github.com> wrote:

> In src/GPSPoint.php
> <https://github.com/zendframework/zend-validator/pull/18#discussion_r34518201>
> :
>
> > +
> > +        list($lat, $long) = explode(',', $value);
> > +
> > +        if ($this->isValidCoordinate($lat, 90.0000) && $this->isValidCoordinate($long, 180.000)) {
> > +            return true;
> > +        }
> > +
> > +        return false;
> > +    }
> > +
> > +    /**
> > +     * @param string $value
> > +     * @param $maxBoundary
> > +     * @return bool
> > +     */
> > +    private function isValidCoordinate($value, $maxBoundary)
>
> Just look around a whole framework and you hardly notice a private
> methods. You need a very good reason to mark method as private. I think
> protected is better.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendframework/zend-validator/pull/18/files#r34518201>.
>